### PR TITLE
[MM-29265] Add end to end tests to the "unarchiveChannelsCmdF" mmctl command

### PR DESCRIPTION
**Pull request by attiss**
#### Summary

This PR adds e2e tests for `unarchiveChannelsCmdF` in the `commands/channel_e2e_test.go` file.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/15758

### CODE REVIEW

**Comment by attiss** [comment]
Note: the happy path test case do not involve testing with the `LocalClient` as the endpoint is not yet migrated to local mode. (See the discussion here: https://community.mattermost.com/core/pl/ot87snk4jf8ujjauc9pewnbmfh)

File: commands/channel_e2e_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+// See LICENSE.txt for license information.
+
+package commands
+
+import (
+	"fmt"
+
+	"github.com/mattermost/mmctl/client"
+	"github.com/mattermost/mmctl/printer"
+	"github.com/spf13/cobra"
```
Comment (ashishbhate) [inline commands/channel_e2e_test.go:11]:
```suggestion
	"fmt"

	"github.com/spf13/cobra"
	
	"github.com/mattermost/mmctl/client"
	"github.com/mattermost/mmctl/printer"
```

Fix for the CI lint error.

**Review by ashishbhate — COMMENTED** [review]
Looking good! Lets add a test case for trying to unarchive an already open channel.

File: commands/channel_e2e_test.go (hunk 1 above)
Comment (attiss) [reply commands/channel_e2e_test.go]: whoops
thanks, fixed!

**Comment by attiss** [comment]
@ashishbhate thanks for the review, updated the PR!

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

File: commands/channel_e2e_test.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		s.Require().Contains(printer.GetErrorLines()[0], fmt.Sprintf("Unable to find channel '%s:%s'", s.th.BasicTeam.Id, "nonexistent-channel"))
+	})
+
+	s.Run("Unarchive open channel", func() {
+		printer.Clean()
+
+		err := unarchiveChannelsCmdF(s.th.SystemAdminClient, &cobra.Command{}, []string{fmt.Sprintf("%s:%s", s.th.BasicTeam.Id, s.th.BasicDeletedChannel.Name)})
+		s.Require().Nil(err)
+		s.Require().Contains(printer.GetErrorLines()[0], fmt.Sprintf("Unable to unarchive channel '%s:%s'", s.th.BasicTeam.Id, s.th.BasicDeletedChannel.Name))
+		s.Require().Contains(printer.GetErrorLines()[0], "Unable to unarchive channel. The channel is not archived.")
```
Comment (ashishbhate) [inline commands/channel_e2e_test.go:54]: This test relies on a side effect of another test (`BasicDeletedChannel` being unarchived in the first test). I suggest not relying on such a side effect, and instead trying to unarchive `BasicChannel`

**Review by ashishbhate — APPROVED** [review]

**Review by ogi-m — APPROVED** [review]
LGTM

File: commands/channel_e2e_test.go (hunk 2 above)
Comment (attiss) [reply commands/channel_e2e_test.go]: Absolutely fair point, thanks for the suggestion!

**Comment by ashishbhate** [comment]
Thanks @attiss!

**Pull request merged**